contrib/repairStatus.py

```diff
@@ -0,0 +1,199 @@
+                                    "query_string": {
+                                        "query": 'host:("' + node + '")'
+                                    }
+                                },
+                                "_cache": True
+                            }
+                        },
+                        {
+                            "fquery": {
+                                "query": {
+                                    "query_string": {
+                                        "query": 'method:("syncComplete")'
+                                    }
+                                },
+                                "_cache": True
+                            }
+                        },
+                        {
+                            "fquery": {
+                                "query": {
+                                    "query_string": {
+                                        "query": 'message:("is fully synced")'
+                                    }
+                                },
+                                "_cache": True
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+    }
+  }
+
+  return json.dumps(query_json)
+
+
+# Execute an ES _count to return the number of ranges that have finished repairing
+def calculate_range_repairs(host, port, node, start_time, current_time):
+  url = 'http://%(host)s:%(port)s/_count' % dict(host=host, port=port)
+  query = generate_range_repair_query(node, start_time, current_time)
+  
+  log.info("Ranges completed: curl %(url)s -d '%(query)s'" % dict(url=url, query=query)) 
+  r = requests.get(url, data=query)
+  
+  if r.status_code == 200:
+    response = json.loads(r.text)
+    return response['count']
+
+
+def generate_total_ranges_query(node, start_time, current_time):
+  query_json = {
+    "fields": "message",
+    "query": {
+      "filtered": {
+        "filter": {
+          "bool": {
+            "must": [
+              {
+                "range": {
+                  "@timestamp": {
+                    "to": current_time,
+                    "from": start_time
+                  }
+                }
+              },
+              {
+                "fquery": {
+                  "_cache": True,
+                  "query": {
+                    "query_string": {
+                      "query": 'host:("' + node + '")'
+                    }
+                  }
+                }
+              }
+            ]
+          }
+        },
+        "query": {
+          "query_string": {
+            "query": 'message: "Starting repair"'
+          }
+        }
+      }
+    }
+  }
+
+  return json.dumps(query_json)
+
+
+# Execute an ES _search to return the total number of ranges that need repairing
+def calculate_total_ranges(host, port, node, start_time, current_time):  
+  url = 'http://%(host)s:%(port)s/_search' % dict(host=host, port=str(port))
```

Comment:
The new-style "format" syntax is a little less error prone than the old-style "%" syntax.  It would look like this:

```python
'http://{host}:{port}/_search'.format(host=host, port=str(port))
```

Notice that you don't need a "s" after the closing braces there - this way the brace-matching in your editor is sufficient to make sure the syntax is right.